keps/sig-cli/3638-built-in-command-shadowing/README.md

```diff
@@ -0,0 +1,685 @@
+previous answers based on experience in the field.
+
+The Troubleshooting section currently serves the `Playbook` role. We may consider
+splitting it into a dedicated `Playbook` document (potentially with some monitoring
+details). For now, we leave it here.
+-->
+Built-in commands and subcommands in `kubectl` always overrule external plugins so that it is not possible to shadow
+already existing built-ins. Thereby, same troubleshooting mechanism should be applied just troubleshooting
+the `kubectl` commands. If the user has a problem when using a plugin, there won't be any change and user needs
+to contact with the plugin owner.
+
+###### How does this feature react if the API server and/or etcd is unavailable?
+
+###### What are other known failure modes?
+
+<!--
+For each of them, fill in the following information by copying the below template:
+  - [Failure mode brief description]
+    - Detection: How can it be detected via metrics? Stated another way:
+      how can an operator troubleshoot without logging into a master or worker node?
+    - Mitigations: What can be done to stop the bleeding, especially for already
+      running user workloads?
+    - Diagnostics: What are the useful log messages and their required logging
+      levels that could help debug the issue?
+      Not required until feature graduated to beta.
+    - Testing: Are there any tests for failure mode? If not, describe why.
+-->
+
+###### What steps should be taken if SLOs are not being met to determine the problem?
+
+## Implementation History
+The KEP was proposed on 2022-10-22
+<!--
+Major milestones in the lifecycle of a KEP should be tracked in this section.
+Major milestones might include:
+- the `Summary` and `Motivation` sections being merged, signaling SIG acceptance
+- the `Proposal` section being merged, signaling agreement on a proposed design
+- the date implementation started
+- the first Kubernetes release where an initial version of the KEP was available
+- the version of Kubernetes where the KEP graduated to general availability
+- when the KEP was retired or superseded
+-->
+
+## Drawbacks
+
+<!--
+Why should this KEP _not_ be implemented?
+-->
+
+## Alternatives
+
+Alternative of the proposed mechanism to overcome this request is to propose a new flag `--use-plugin`. User can
+pass this flag when they want to explicitly use the external plugin. Thanks to that, user also can shadow
+already existing subcommands. For example `kubectl create job` executes built-in `job` subcommand, on the other hand
+`kubectl create job --use-plugin` executes custom `kubectl-create-job` plugin.
+
+However, downside of this alternative is that usability hurts without any gain;
+- Everytime user has to pass a new flag to execute external plugin. 
+- This flag can not be passed to external plugins and this makes `--use-plugin` flag a special flag.
+- There is no clear use case allowing plugins for already existed subcommands which means that we want to disable this 
```

Comment:
Possibly for migration if there's a conflict later?
Though the user could also call the plugin by the real binary name presumably.

Reply:
Yes, I think if user really needs to use plugin, he/she can easily execute manually